### DISCUSSION TITLE
V3 Issue with Alpine - variables not found.

### DISCUSSION THREAD

**Discussion by edalzell** [livewire/livewire#5860] [7. Bugs (archived)]
I have a component with some properties:

```php
    public bool $show = false;
```

In that component's blade file, I use alpine to show/hide:

```js
    x-data="{
        show: @entangle($attributes->wire('model')),
```

That is called like so:

```html
<x-modal.dialog wire:model="show">
```

When I toggle that `show` property in Livewire, on the front end I get these errors:
<img width="990" alt="CleanShot 2023-07-21 at 17 21 18@2x" src="https://github.com/livewire/livewire/assets/6069653/14ce5df7-b8de-4d97-bd20-7409dca4a041">

Repo here: https://github.com/edalzell/lw-bug

**Comment by edalzell** [comment]
same issue occurs if I try `$wire.$entangle($attributes->wire('model'))`

**Reply by lrljoe** [reply]
Just so I'm clear here.

You have a Livewire Component.
The component has the following:
``` 
    public bool $show = false;
```
That Livewire Component has a blade, which contains 
```
<x-modal.dialog wire:model="show">
```
Then in your 
modal.dialog.blade.php

You have something like this:
```
<div x-data="{ show: @entangle($attributes->wire('model')) }">
       <div x-show="show">
        Test
    </div>

</div>
```

If so - that is definitely working for me.

**Reply by lrljoe** [reply]
Personally, I probably wouldn't start passing things back and forth so deeply, when you could just set a "dialogShow" attribute further up.
A modal to my mind is a very good candidate for a nested component if you are going to do work with the data at least.

**Reply by edalzell** [reply]
What do you mean “passing things back and forth so deeply”? The repo is above, showing it all.

**Reply by edalzell** [reply]
Interesting that it works for you though, wonder what else I’m doing that’s mucking it up.

**Comment by edalzell** [comment]
OK, @joshhanley did some wonderful debugging and found it was some empty `div`s in a view that cause the issue. He's adding a failing test so that it can get addressed at some point.

**Reply by joshhanley** [reply]
Here is the PR #5889 with failing test.
